classes/db/DAOResultFactory.php

```diff
@@ -259,6 +259,22 @@ public function toAssociativeArray($idField = 'id')
         }
         return $returner;
     }
+
+    /**
+     * Convert this iterator to a collection.
+     *
+     * @return Collection<T>
+     */
+    public function toCollection(): Collection
+    {
+        $retCollection = collect();
+
+        while ($object = $this->next()) {
+            $retCollection->push($object);
```

Comment:
This will cause a big result set to be instantiated in memory all at once; I'd suggest using a generator instead. https://josephsilber.com/posts/2020/07/29/lazy-collections-in-laravel#lazy-collections-wrap-a-generator-function

Reply:
Maybe something like that 

```
public function toLazyCollection(): LazyCollection
{
    return LazyCollection::make(function () {
        while ($object = $this->next()) {
            yield $object;
        }
    });
}
```
In any case I removed the function all together because it is not referenced anywhere in the code anymore.